spirv_hlsl.cpp

```diff
@@ -1238,7 +1272,16 @@ void CompilerHLSL::emit_texture_op(const Instruction &i)
 void CompilerHLSL::emit_uniform(const SPIRVariable &var)
 {
 	add_resource_name(var.self);
-	statement(variable_decl(var), ";");
+	auto &type = get<SPIRType>(var.basetype);
+	if (options.shader_model >= 40 && (type.basetype == SPIRType::Image || type.basetype == SPIRType::SampledImage))
+	{
+		statement("Texture2D<float4> ", to_name(var.self), ";");
```

Comment:
Is it possible to move this code into emit_uniform()? iirc, CompilerHLSL overrides that.